Smt/Util.lean

```diff
@@ -99,6 +99,10 @@ partial def unfoldProjInsts? (e : Expr) : MetaM (Option Expr) := do
 /-- Returns an expression equivalent to `e` with all typeclass projections
     unfolded. -/
 def unfoldAllProjInsts : Expr → MetaM Expr :=
-  Meta.transform (pre := fun e => return (← unfoldProjInsts? e).getD e |> .visit)
+  Meta.transform (pre := pre)
+where
+  pre e : MetaM TransformStep := return match ← Meta.unfoldProjInst? e with
+    | some e => .visit e
+    | none   => .continue
```

Comment:
Did you intend the change in behaviour here? We used to do `| some e' => visit e' | none => visit e`.

Reply:
Yup! Pretty cool if we exclude the amount of time it took me to understand why Lean was suddenly running into infinite loops 😂